util_test.go

```diff
@@ -0,0 +1,46 @@
+package libjavabuildpack_test
+
+import (
+	"testing"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"path/filepath"
+	"github.com/cloudfoundry/libjavabuildpack/test"
+	"os"
+	"github.com/cloudfoundry/libjavabuildpack"
+)
+
+func TestExtractTarGz(t *testing.T) {
+	spec.Run(t, "ExtractTarGz", testExtractTarGz, spec.Report(report.Terminal{}))
+}
+
+func testExtractTarGz(t *testing.T, when spec.G, it spec.S) {
+	var outDir string
+
+	it.Before(func() {
+		outDir = test.ScratchDir(t, "out")
+	})
+
+	it.After(func() {
+		os.RemoveAll(outDir)
+	})
+
+	it("extracts the archive", func() {
+		err := libjavabuildpack.ExtractTarGz(filepath.Join("fixtures", "test-archive.tar.gz"), outDir, 0)
```

Comment:
You can't depend on the working directory of the test invocation so pointing at `fixtures` relative is problematic.  You'll need to [find the Buildpack root](https://github.com/cloudfoundry/libjavabuildpack/blob/master/test/test_helpers.go#L109) and work from there.  Probably worthwhile to add a `test_helper` called something like `FixturePath(t, "test-archive.tar.gz")` since I think this will be quite common.